component/src/main/java/org/wso2/extension/siddhi/io/sqs/source/SQSSourceConfig.java

```diff
@@ -0,0 +1,162 @@
+                    super.setAccessKey(optionHolder.validateAndGetStaticValue(key));
+                    break;
+                case SQSConstants.SECRET_KEY_NAME:
+                    super.setSecretKey(optionHolder.validateAndGetStaticValue(key));
+                    break;
+                case SQSConstants.REGION_NAME:
+                    super.setRegion(optionHolder.validateAndGetStaticValue(key));
+                    break;
+                case SQSConstants.WAITING_TIME_NAME:
+                    this.setWaitTime(getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_WAITING_TIME));
+                    break;
+                case SQSConstants.VISIBILITY_TIMEOUT_NAME:
+                    this.setVisibilityTimeout(
+                            getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_VISIBILITY_TIMEOUT));
+                    break;
+                case SQSConstants.MAX_NUMBER_OF_MESSAGES_NAME:
+                    this.setMaxNumberOfMessages(
+                            getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_MAX_NUMBER_OF_MESSAGES));
+                    break;
+                case SQSConstants.PURGE_MESSAGES_NAME:
+                    this.setDeleteAfterConsume(
+                            getBooleanValue(optionHolder, key, SQSConstants.DEFAULT_DELETE_AFTER_CONSUME));
+                    break;
+                case SQSConstants.DELETE_RETRY_INTERVAL_NAME:
+                    this.setRetryInterval(
+                            getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_RETRY_INTERVAL));
+                    break;
+                case SQSConstants.MAX_NUMBER_OF_DELETE_RETRY_ATTEMPTS_NAME:
+                    this.setRetryCountLimit(
+                            getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_RETRY_COUNT_LIMIT));
+                    break;
+                case SQSConstants.POLLING_INTERVAL_NAME:
+                    this.setPollingInterval(
+                            getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_POLLING_INTERVAL));
+                    break;
+                case SQSConstants.PARALLEL_CONSUMERS_NAME:
+                    this.setThreadPoolSize(
+                            getIntegerOptionValue(optionHolder, key, SQSConstants.DEFAULT_PARALLEL_CONSUMERS));
+                    break;
+                default:
+                    // not a supported option.
+                    break;
+            }
+        });
+    }
+
+    public String[] getRequestedTransportProperties() {
+        return requestedTransportProperties.clone();
+    }
+
+    public int getWaitTime() {
+        return waitTime;
+    }
+
+    public void setWaitTime(int waitTime) {
+        this.waitTime = waitTime;
+    }
+
+    public int getVisibilityTimeout() {
+        return visibilityTimeout;
+    }
+
+    public void setVisibilityTimeout(int visibilityTimeout) {
+        this.visibilityTimeout = visibilityTimeout;
```

Comment:
is visibilityTimeout accessible from outside? If not, remove this setter and assign to the variable in the constructor.